### ISSUE TITLE
I18n.l('time.formats.short', undefined) behavior (raise an error)

### ISSUE THREAD

**Issue by pironim** [fnando/i18n-js#535]
I18n.l('time.formats.short', undefined) raise and error
on the other hands
I18n.l('time.formats.short', '') returns 'Invalid Date'
I check it for version 3.2.0 

As for me it's good to have same or similar behavior for unknown data.
real world example:
I'm trying to render view for rail model and it does not have create_at and updated_at field.
view crashing for 
I18n.l('time.formats.short', createdAt)

I can prepare patch if it's acceptable.
Possible scenario: undefined can return 'Incorrect Date' or empty string or maybe current date

**Comment by PikachuEXE** [comment]
I guess the error is due to this line
https://github.com/fnando/i18n-js/blob/master/app/assets/javascripts/i18n.js#L923

PR is welcome
I think the line should be something like
```js
// `date` can be any value, null, undefined, non Date objects
// We check `typeof getTime` to ensure no error 
if (date == null || (! (typeof date.getTime === "function")) || isNaN(date.getTime())) {
```

If you are talking about other function please let me know (I don't know the JS part well enough

**Comment by pironim** [comment]
@PikachuEXE Looks like you propose more detailed check to raise an error.
I want to achieve opposite thing and avoid error for `undefined` and maybe `null` value
undefined and null can be returned as is. what do you think?

**Comment by pironim** [comment]
something like that
```
it("don't localizes empty values", function(){
   I18n.locale = "pt-BR";
   expect(I18n.l("date.formats.default", "")).toEqual("Invalid Date");
   expect(I18n.l("date.formats.default", undefined)).toEqual(undefined);
   expect(I18n.l("date.formats.default", null)).toEqual(null);
});
```

^ specs example

**Comment by PikachuEXE** [comment]
I think we should
- Return "Invalid Date" for all invalid values
- Add option to return input value as is for invalid values

```js
it("don't localizes empty values", function(){
   I18n.locale = "pt-BR";
   expect(I18n.l("date.formats.default", "")).toEqual("Invalid Date");
   expect(I18n.l("date.formats.default", undefined)).toEqual("Invalid Date");
   expect(I18n.l("date.formats.default", null)).toEqual("Invalid Date");

   // Option name and value to be decided later 
   expect(I18n.l("date.formats.default", "", on_invalid_datetime: "return_input")).toEqual("");
   expect(I18n.l("date.formats.default", undefined, on_invalid_datetime: "return_input")).toEqual(undefined);
   expect(I18n.l("date.formats.default", null, on_invalid_datetime: "return_input")).toEqual(null);
});
```

**Comment by pironim** [comment]
Invalid Date belongs to vanila js and result of operation like
```
(new Date('')).toString()
"Invalid Date"
```

Simulating this behavior questionable to me.
as a result on_invalid_datetime: for this scenario 
looks like **simulate Invalid Date** or return as is

**'Invalid Date'** vs **undefined**
**'Invalid Date'** vs **null**

looks like almost identical
additional flag just increase complexity and it's hard to imagine real use case for **'Invalid Date'** vs **null**

I'm trying to convince you that same value for special values much simpler and should not brake anything.
currently we just have silly error like toString can't be called

**Comment by PikachuEXE** [comment]
Just fixed in https://github.com/fnando/i18n-js/commit/869d1689ed788ff50121de492db354652971c23d
See whether that's what you want

**Comment by pironim** [comment]
@PikachuEXE Yep. Looks similar to my changes on local machine.

**Comment by pironim** [comment]
![200w_s](https://user-images.githubusercontent.com/1135304/57367936-a9d54100-7192-11e9-9616-1c0c606a2c6e.gif)
Thank you

**Comment by PikachuEXE** [comment]
Released as [`3.2.2`](https://github.com/fnando/i18n-js/releases/tag/v3.2.2)
Please close this if it's fixed :)
